### PR TITLE
TRIM and CHARSET tests, bug fixes.

### PR DESCRIPTION
**Pull request by codebybrett**
This adds bug fixes, modifications and tests for TRIM and CHARSET.

The argument to TRIM/WITH was used simply for creating a BITSET! which is the purpose of CHARSET. Maybe originally this was a choice of convenience so that a console user did not have to call CHARSET themselves.

To reduce duplication, the work of creating a BITSET! within TRIM is delegated to CHARSET, which means moving CHARSET from mezz-series.r to base-series.r

In keeping with the idea of convenience, BITSET! becomes an allowed argument type for /WITH.

### CODE REVIEW

**Comment by codebybrett** [comment]
I simplified the strategy for making the bitset in CHARSET. What I don't know is if make bitset! uses the argument as a hint for the size of bitset it should create. If it does, I just inadvertently dropped it.

File: src/mezz/base-series.r [hunk 1]
```diff
@@ -233,7 +255,27 @@ trim: function [
...
+                    auto
+                    any [head_TRIM tail_TRIM lines]
+                ]
+                all [
+                    any [all_TRIM with]
+                    any [auto head_TRIM tail_TRIM lines]
+                ]
+            ][
+                fail "Invalid refinements for TRIM of STRING!"
+            ]
```
Comment (hostilefork) [inline src/mezz/base-series.r]: Continued food for thought on the place of things like AND, OR, THEN...as I want people trying them and getting experience with them:

    any [
        auto and (any [head_TRIM tail_TRIM lines])
        any [all_TRIM with] and (any [auto head_TRIM tail_TRIM lines])
    ] then [
        fail "Invalid refinements for TRIM of STRING!"
    ]

Does the mixture make the intent more clear?  Wondering what kinds of thoughts it brings about.

File: src/mezz/base-series.r [hunk 2]
```diff
@@ -243,23 +285,11 @@ trim: function [
...
-                    for-each b str [append array b]
-                    make bitset! array
-                ]
-                with [make bitset! str]
-            ] else [#{00}]
+            rule: either with [
+                either bitset? str [str] [charset str]
+            ][
+                #{00}
+            ]
```
Comment (hostilefork) [inline src/mezz/base-series.r]: I'm curious how comfortable people can get with ?? and !!, and mixing them with other branching constructs:

    rule: if with [either bitset? str [str] [charset str]] !! #{00}

If one warms up to them, I think they are pretty nice.

**Comment by hostilefork** [comment]
My comments were just thinking points--for now, rebased and merged as is!  Thanks!

https://github.com/metaeducation/ren-c/commit/7bee422832730d12ee53330bb20d18bd494dd3e5

**Comment by codebybrett** [comment]
I did try ?? !!  (which is where the || issue came from - typo), and I tried **and**.

But my use of them caused errors, which I couldn't immediately work out, so I went with old style. I probably should have returned to them once I got the code working.

**Pull request closed without merging**